### PR TITLE
Add test

### PR DESCRIPTION
**Pull request by davidbrochart**

### CODE REVIEW

File: tests/test_reactive.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return age
+
+
+def test(capfd):
+    foo = Foo()
+
+    foo.name = "John"
+    out, err = capfd.readouterr()
+    assert out == dedent("""\
+        old='Paul', new='Paul'
```
Comment (davidbrochart) [inline tests/test_reactive.py:33]: The `watch_` method is called a first time with the same value.
I'm not sure it's the case in Textual @willmcgugan?

File: tests/test_reactive.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    out, err = capfd.readouterr()
+    assert out == dedent("""\
+        old='John', new='Steve'
+    """)
+    
+    foo.age
+    out, err = capfd.readouterr()
+    assert out == dedent("""\
+        age=33
+        age=33
```
Comment (davidbrochart) [inline tests/test_reactive.py:49]: Same here, the `compute_` method is called twice the first time.

File: tests/test_reactive.py (hunk 1 above)
Comment (willmcgugan) [reply tests/test_reactive.py:33]: It's not called first time, unless `init=True` IIRC

Comment (davidbrochart) [reply tests/test_reactive.py:33]: Indeed, thanks. Any reason for having `init=True` as the default value?

Comment (willmcgugan) [reply tests/test_reactive.py:33]: Generally in a Textual app, you want to respond to changes to the state. But there isn’t often a need to do anything when you first set the state.

It’s also problematic if your watcher makes assumptions about other widgets, that may not yet have updated their state. So init=False was the safer default.

Comment (davidbrochart) [reply tests/test_reactive.py:33]: But it's the [opposite](https://github.com/Textualize/textual/blob/10e61987e36a45b8a174d179f86e2280e66cfb94/src/textual/reactive.py#L313), right?

Comment (willmcgugan) [reply tests/test_reactive.py:33]: Ah, you’re right. Don’t recall why we changed it. Suspect it more aligned with user expectations.

**Pull request merged**